File: go/test/models_test.go
package test

import (
	"math"
	"shapes/test/fixtures"
	"testing"

	"github.com/stretchr/testify/assert"
)

const DELTA = 0.001

func TestCircle(t *testing.T) {
	t.Run("Test Circle Area", func(t *testing.T) {
		circle := fixtures.SetupCircle()
		result := circle.Area()
		expected := 754.767
		assert.True(t, math.Abs(expected-result) < DELTA)
	})

	t.Run("Test Circle Circumference", func(t *testing.T) {
		circle := fixtures.SetupCircle()
		result := circle.Perimeter()
		expected := 97.389
		assert.True(t, math.Abs(expected-result) < DELTA)
	})
}

func TestRectangle(t *testing.T) {
	t.Run("Test Rectangle Area", func(t *testing.T) {
		rectangle := fixtures.SetupRectangle()
		result := rectangle.Area()
		expected := 756.437
		assert.True(t, math.Abs(expected-result) < DELTA)
	})

	t.Run("Test Rectangle Perimeter", func(t *testing.T) {
		rectangle := fixtures.SetupRectangle()
		result := rectangle.Perimeter()
		expected := 112.0
		assert.True(t, math.Abs(expected-result) < DELTA)
	})
}

func TestSquare(t *testing.T) {
	t.Run("Test Square Area", func(t *testing.T) {
		square := fixtures.SetupSquare()
		result := square.Area()
		expected := 105.062
		assert.True(t, math.Abs(expected-result) < DELTA)
	})

	t.Run("Test Square Perimeter", func(t *testing.T) {
		square := fixtures.SetupSquare()
		result := square.Perimeter()
		expected := 41.0
		assert.True(t, math.Abs(expected-result) < DELTA)
	})
}

File: ruby/test/models_test.rb
# frozen_string_literal: true

require "minitest/autorun"
require "minitest/reporters"

require_relative "fixtures/helper"

Minitest::Reporters.use! Minitest::Reporters::SpecReporter.new(color: true)

DELTA = 0.001

class TestCircle < Minitest::Test
    def test_circle_area
        circle = Helper.setup_circle
        result = circle.area
        expected = 754.767
        assert((expected - result).abs < DELTA)
    end

    def test_circle_perimeter
        circle = Helper.setup_circle
        result = circle.perimeter
        expected = 97.389
        assert((expected - result).abs < DELTA)
    end
end

class TestRectangle < Minitest::Test
    def test_rectangle_area
        rectangle = Helper.setup_rectangle
        result = rectangle.area
        expected = 756.437
        assert((expected - result).abs < DELTA)
    end

    def test_rectangle_perimeter
        rectangle = Helper.setup_rectangle
        result = rectangle.perimeter
        expected = 112.0
        assert((expected - result).abs < DELTA)
    end
end

class TestSquare < Minitest::Test
    def test_square_area
        square = Helper.setup_square
        result = square.area
        expected = 105.062
        assert((expected - result).abs < DELTA)
    end

    def test_square_perimeter
        square = Helper.setup_square
        result = square.perimeter
        expected = 41.0
        assert((expected - result).abs < DELTA)
    end
end
